File: backend/src/main/java/com/tfg/backend/domain/Dance.java
package com.tfg.backend.domain;

import jakarta.persistence.*;
import lombok.*;

import java.util.List;

@Entity
@Table(name = "dances",
        uniqueConstraints =
            @UniqueConstraint(columnNames = "name"))
@Getter
@Setter
public class Dance {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String name;

    private String region;

    @Column(length = 2000)
    private String description;

    private String videoUrl;

    @Column(columnDefinition = "TEXT")
    private String history;
    @Column(columnDefinition = "TEXT")
    private String origin;
    @Column(columnDefinition = "TEXT")
    private String clothing;
    @Column(columnDefinition = "TEXT")
    private String musicCharacteristics;
    @Column(columnDefinition = "TEXT")
    private String danceSteps;

    @OneToMany(mappedBy = "dance")
    private List<UserDanceProgress> progresses;

    public Dance() {}

    public Dance(String name, String region, String description, String videoUrl, String history, String origin, String clothing, String musicCharacteristics) {
        this.name = name;
        this.region = region;
        this.description = description;
        this.videoUrl = videoUrl;
        this.history = history;
        this.origin = origin;
        this.clothing = clothing;
        this.musicCharacteristics = musicCharacteristics;
    }
}

File: frontend/lib/models/dance.dart
class Dance {
  final int id;
  final String name;
  final String region;
  final String description;
  final String videoUrl;
  final String history;
  final String origin;
  final String clothing;
  final String musicCharacteristics;

  Dance({
    required this.id,
    required this.name,
    required this.region,
    required this.description,
    required this.videoUrl,
    required this.history,
    required this.origin,
    required this.clothing,
    required this.musicCharacteristics
  });

  factory Dance.fromJson(Map<String, dynamic> json) {
    return Dance(
      id: json['id'],
      name: json['name'] ?? '',
      region: json['region'] ?? '',
      description: json['description'] ?? '',
      videoUrl: json['videoUrl'] ?? '',
      history: json['history'] ?? '',
      origin: json['origin'] ?? '',
      clothing: json['clothing'] ?? '',
      musicCharacteristics: json['musicCharacteristics'] ?? ''
    );
  }
}